kroxylicious-sample/src/test/java/io/kroxylicious/sample/SampleFilterIntegrationTest.java

```diff
@@ -0,0 +1,154 @@
+import io.kroxylicious.testing.kafka.junit5ext.KafkaClusterExtension;
+
+import static io.kroxylicious.test.tester.KroxyliciousConfigUtils.proxy;
+import static io.kroxylicious.test.tester.KroxyliciousConfigUtils.withDefaultFilters;
+import static io.kroxylicious.test.tester.KroxyliciousTesters.kroxyliciousTester;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@ExtendWith(KafkaClusterExtension.class)
+public class SampleFilterIntegrationTest {
+
+    private static final String PRE_TRANSFORM_VALUE = "foo";
+    private static final String NO_TRANSFORM_VALUE = "sample";
+    private static final String PRODUCE_TRANSFORM_VALUE = "bar";
+    private static final String FETCH_TRANSFORM_VALUE = "baz";
+    private static final String FIND_CONFIG_FIELD = "findValue";
+    private static final String REPLACE_CONFIG_FIELD = "replaceValue";
+    private static final Map<String, Object> PRODUCE_CONFIG = Map.of(FIND_CONFIG_FIELD, "foo", REPLACE_CONFIG_FIELD, "bar");
+    private static final Map<String, Object> FETCH_CONFIG = Map.of(FIND_CONFIG_FIELD, "bar", REPLACE_CONFIG_FIELD, "baz");
+    private static final Integer TIMEOUT_SECONDS = 10;
+    private static final String TOPIC_NAME = "test";
+    private static final Integer TOPIC_PARTITIONS = 1;
+    private static final Short TOPIC_REPLICATION = 1;
+
+    KafkaCluster cluster;
+    private KroxyliciousTester tester;
+
+    @BeforeEach
+    public void beforeEach() {
+        tester = kroxyliciousTester(withDefaultFilters(proxy(cluster))
+                .addToFilters(new FilterDefinitionBuilder(SampleContributor.SAMPLE_PRODUCE).withConfig(PRODUCE_CONFIG).build())
+                .addToFilters(new FilterDefinitionBuilder(SampleContributor.SAMPLE_FETCH).withConfig(FETCH_CONFIG).build()));
+    }
+
+    @AfterEach
+    public void afterEach() {
+        tester.close();
+    }
+
+    @Test
+    public void sampleFilterWillTransformRoundTripTest(Admin admin) {
+        try (Producer<String, String> producer = tester.producer();
+                Consumer<String, byte[]> kafkaClusterConsumer = new KafkaConsumer<>(mergeMaps(cluster.getKafkaClientConfiguration(),
+                        Map.of(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class, ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+                                ByteArrayDeserializer.class, ConsumerConfig.GROUP_ID_CONFIG, "group-id-0", ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")));
+                Consumer<String, byte[]> proxyConsumer = tester.consumer(Serdes.String(), Serdes.ByteArray(),
+                        Map.of(ConsumerConfig.GROUP_ID_CONFIG, "group-id-1", ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest"))) {
+            admin.createTopics(List.of(new NewTopic(TOPIC_NAME, TOPIC_PARTITIONS, TOPIC_REPLICATION))).all().get(TIMEOUT_SECONDS, TimeUnit.SECONDS);
```

Comment:
Should we do this in setup?

Reply:
Alternatively I'd be looking at extracting methods for the whole test method like:

```
givenTopic(TOPIC_NAME, TOPIC_PARTITIONS, TOPIC_REPLICATION)
whenProduce(TOPIC_NAME, PRE_TRANSFORM_VALUE)
thenOnlyRecordConsumedFromTopicEquals(kafkaClusterConsumer, TOPIC_NAME, x, y, z)
```
(not exact naming but could make it easier to grok)

since they're quite big

Reply:
Good idea! I've broken down each "section" of these test methods and split them out into something a bit more readable.